### ISSUE TITLE
segmentation fault in python-tk@3.12 and python-tk@3.13

### ISSUE THREAD

**Issue by cosmosgenius** [Homebrew/homebrew-core#199354]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 4.4.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e78a0adb4f260913fffdc956cc8c20fb54109feb
Last commit: 3 days ago
Branch: stable
Core tap JSON: 28 Nov 07:19 UTC
Core cask tap JSON: 28 Nov 07:19 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 16.0.0 build 1600
Git: 2.47.1 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false

Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  packer
  vault

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  numpy
```

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I was trying to compile and run koboldcpp https://github.com/LostRuins/koboldcpp.

### What happened (include all command output)?

The following python snippet segfaults. This for both `python-tk@3.12` and `python-tk@3.13`
```python
import tkinter
import threading

original_windowwidth = 580
original_windowheight = 560
windowwidth = original_windowwidth
windowheight = original_windowheight

root = tkinter.Tk()
root.geometry(str(windowwidth) + "x" + str(windowheight))
gpu_choice_var = tkinter.StringVar(value="1")
quick_gpuname_label = tkinter.Label(root, text="hello")

def outPrint(*args):
    quick_gpuname_label.configure(text="1")
    print(args)
gpu_choice_var.trace_add("write", outPrint)

gpuinfo_thread = threading.Thread(target=outPrint)
gpuinfo_thread.start()

# gpu_choice_var.set("1")
# outPrint()
# root.mainloop()
```

### What did you expect to happen?

segmentation fault to not happen.

This problem came up due to the recent change from tcl 8.x to tcl 9.0.0 as the default tcl package.
I am not sure where in the segfault issue is from
PFA of segfault dump

[segfault.txt](https://github.com/user-attachments/files/17945168/segfault.txt)

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install python@3.12 python-tk@3.12
python3.12 <above script>
```

**Comment by vitaminniy** [comment]
Hello and sorry for reopening the issue but it happens for me as well while installing `python@3.13`.

```bash
➜ ~ brew config
HOMEBREW_VERSION: 4.4.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3001afe88112c13b5aafdd92e097680897060881
Last commit: 4 days ago
Branch: stable
Core tap JSON: 19 Dec 20:12 UTC
Core cask tap JSON: 19 Dec 20:12 UTC
HOMEBREW_PREFIX: /Users/vitaminniy
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: dodeca-core 64-bit arm_lobos
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false

➜  ~ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Your Homebrew's prefix is not /opt/homebrew.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

**Comment by cho-m** [comment]
Needs a fix on CPython side. Opened https://github.com/python/cpython/pull/128103 as potential way of handling.

**Comment by cho-m** [comment]
For Python 3.13, can try `brew reinstall python-tk@3.13` as rebuilt bottle with some fixes in #201830.

Still need to handle Python 3.12.
